src/components/inputs/FilterButtonsGroupSelector/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { FilterButtons } from '@/components/buttonsGroup/FilterButtonsGroup'
+import { DropdownIcon } from '@/icons'
+import { Box, ClickAwayListener, Popper, Stack, Typography } from '@mui/material'
+import { useState } from 'react'
+
+export const FilterButtonsGroupSelector = ({
+  filterButtons,
+  activeButtonIndex,
+}: {
+  filterButtons: FilterButtons[]
+  activeButtonIndex: number
+}) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+  const open = Boolean(anchorEl)
+  const id = open ? 'filterButtons-selector-popper' : ''
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    event.stopPropagation()
+    if (event.key === 'Escape') {
+      setAnchorEl(null)
+    }
+  }
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    setAnchorEl(anchorEl ? null : event.currentTarget)
+  }
```

Comment:
```suggestion
  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    setAnchorEl(prev => prev ? null : event.currentTarget)
  }
```

This helps avoid bugs related to quick clicking / double clicks since anchorEl could be stale